salt/utils/master.py

```diff
@@ -189,16 +189,16 @@ def _get_minion_grains(self, *minion_ids, **kwargs):
             cret = dict([(minion_id, mcache) for (minion_id, mcache) in six.iteritems(cached_grains) if mcache])
             missed_minions = [minion_id for minion_id in minion_ids if minion_id not in cret]
             log.debug('Missed cached minion grains for: %s', missed_minions)
-            if self.grains_fallback:
+            if self.grains_fallback and len(missed_minions) > 0:
                 lret = self._get_live_minion_grains(missed_minions)
-            ret = dict(list(six.iteritems(dict([(minion_id, {}) for minion_id in minion_ids]))) + list(lret.items()) + list(cret.items()))
+            ret = dict(list(six.iteritems(dict([(minion_id, {}) for minion_id in minion_ids]))) + list(cret.items()) + list(lret.items()))
```

Comment:
can you convert this to using dict interpretation instead of dict()?

so instead of

`dict([x, y] for x, y in z.items())`

you would use 

`{x: y for x, y in z.items()}`

we no longer support python 2.6 which is what the dict() wrapper was used for.

Reply:
same for this one.